### ISSUE TITLE
grouped fields list only the first option as "required". Form validation: 

### ISSUE THREAD

**Issue by StacyBleeks** [wet-boew/wet-boew#9083]
there is an issue with marking up the group header, and then only the 1st option of a form list as "required". Of course it is important to know what field is "requried". 
these examples: "favourite pet" and "citizenship type" are marked up as "requried" within the group header. However, the 1st option in the list is **also** marked up as "requried". 
This is confusing as it could skew the data. In other words, since the **other** options within the same list are not marked up as "required" the respondent might be lead to believe, that since the 1st option is the only one labeled as 'required", this could be the one the respondent will be inclined to select.

Even the github homepage only offers the group header as a required field (as opposed to also marking up the 1st option as 
 "required".
You will notice that for the dropdown/combo box labelled “what do you need help with”, only the overarching group header is marked up as required, not all or any of the options within the dropdown.

See what I mean here:
https://support.github.com/contact/account

So, either all of the options should be marked up as "required" or only the group header. Maybe the options could use  ARIA describedby (or labbeledby)  to reference the "required" aspect of the group header. 
In any case, only marking up the 1st option as "required" seems to pose an issue.
 

 "

**Comment by StacyBleeks** [comment]
@duboisp What are your thoughts?

**Comment by StacyBleeks** [comment]
there is an issue with marking up the group header, and then only the 1st option of a form list as "required".  Of course it is important to know what field is "required". 
these examples: "favourite pet" and "citizenship type" are marked up as "required" within the group header. However, the 1st option in the list is **also** marked up as "required". 
This is confusing as it could skew the data. In other words, since only the 1st option, within the same list is marked up as "required" the respondent might be lead to believe, that since the 1st option is the only one labeled as 'required", this could be the one the respondent will be inclined to select.

Even the github homepage only offers the group header as a required field (as opposed to also marking up the 1st option as 
 "required".
You will notice that for the dropdown/combo box labelled “what do you need help with”, only the overarching group header is marked up as required, not all or any of the options within the dropdown.

See what I mean here:
https://support.github.com/contact/account

So, either all of the options should be marked up as "required" or only the group header. Maybe the options could use  ARIA describedby (or labbeledby)  to reference the "required" aspect of the group header. 
In any case, only marking up the 1st option as "required" poses an issue.
 



 "

**Comment by duboisp** [comment]
@StacyBleeks yes I agree this is confusing. It is possible to update the form validation plugin to enhance the markup for the grouped fields list. The first step would be to create a code sample of the the enhanced markup with the ARIA properties you have proposed.

How the following markup should be enhanced
```html
<fieldset class="chkbxrdio-grp">
	<legend class="required"><span class="field-name">Favourite pets</span> <strong class="required">(required)</strong></legend>
	<div class="checkbox">
		<label for="animal1"><input type="checkbox" name="animal" value="1" id="animal1" required="required" />&#160;&#160;Dog</label>
	</div>
	<div class="checkbox">
		<label for="animal2"><input type="checkbox" name="animal" value="2" id="animal2" />&#160;&#160;Cat</label>
	</div>
	<div class="checkbox">
		<label for="animal3"><input type="checkbox" name="animal" value="3" id="animal3" />&#160;&#160;Fish</label>
	</div>
	<div class="checkbox">
		<label for="animal4"><input type="checkbox" name="animal" value="4" id="animal4" />&#160;&#160;Other</label>
	</div>
</fieldset>
```

**Comment by StacyBleeks** [comment]
@duboisp thanks very much for your quick reply. I noticed you added some features to this issue, does this mean it will be assigned to someone? What is the next step towards resolving the issue?

**Comment by StacyBleeks** [comment]
@duboisp  I don't think ARIA is the way to go. I think it should simply be the group header that is marked up as required. This way the "required attribute will apply to all other parts of it's form element. 
Again, the GitHub example provided on their home page is a perfect example. 

In other words, no "required ID should be part of the markup for the options within a dropdown combo box. Offering the "required" attribute at the group level works well. It serves the purpose in a clear concise way without imposing any ambiguity or confusion.

**Comment by thekodester** [comment]
What about using data attribute feature of the jQuery Validation plugin?

Here a demo page for testing to see if this makes a difference: https://thekodester.ca/wet/demos/forms/boxes

```
<fieldset class="chkbxrdio-grp">
<legend class="required"><span class="field-name">Favourite pets</span> <strong class="required">(required)</strong></legend>
<div class="checkbox">
<label for="animal1"><input type="checkbox" name="animal" value="1" id="animal1" data-rule-required="true">&#160;&#160;Dog</label>
</div>
<div class="checkbox">
<label for="animal2"><input type="checkbox" name="animal" value="2" id="animal2">&#160;&#160;Cat</label>
</div>
<div class="checkbox">
<label for="animal3"><input type="checkbox" name="animal" value="3" id="animal3">&#160;&#160;Fish</label>
</div>
<div class="checkbox">
<label for="animal4"><input type="checkbox" name="animal" value="4" id="animal4">&#160;&#160;Other</label>
</div>
</fieldset>
```

**Comment by StacyBleeks** [comment]
@thekodester  Your data attribute feature of the jQuery Validation plugin works really well from a screen reader user perspective. Thank you for sharing your work. What would need to happen in order to move forward with this as a replacement example? If I was able to provide confrimation that your new example works well using various screen reader/ browser combinations, would it be possible to move forward with your new example to replace the current example found here:
(https://wet-boew.github.io/v4.0-ci/demos/formvalid/formvalid-en.html).

**Comment by RubyDo** [comment]
I tested @thekodester 's fix on NVDA with Chrome and it works fine. 
Here is a demo for before and after 
https://www.youtube.com/watch?v=CUNfjUzBLQ0

**Comment by StacyBleeks** [comment]
@RubyDo  thank you! I have also tested the @thekodester  jQuery solution using JAWS & Chrome, JAWS & Firefox, JAWS & IE11, JAWS & Edge, Safari & VoiceOver and all work well.
 @duboisp  what do you think?

**Comment by thekodester** [comment]
It would be a really simple PR. We need to add some explanation why using data-rule is better vs other method.

**Comment by thekodester** [comment]
Submitting PR for this. French translation we'll be required for: "It is recommended to use data-rule-required="true" vs required="required" as this avoids screen readers indicating that the first option is required."

<img width="1183" alt="Screen Shot 2021-05-04 at 12 26 33 AM" src="https://user-images.githubusercontent.com/23484460/116960770-67ef1080-ac6f-11eb-9a8e-379c264eb0b5.png">

**Comment by StacyBleeks** [comment]
Should we add something to make the point: If only the first option is being identified as required for screen reader users, it will offer an imbalanced design and could lead some respondents to be inclined to choose the first options. The data-rule-required="true approach provides a streamlined design offering the required"  aspect at the group level.

**Comment by RobJohnston** [comment]
Bonus if this fixes https://github.com/wet-boew/wet-boew/issues/6901 for users with JavaScript turned off.

**Comment by thekodester** [comment]
@StacyBleeks the text description can be change to your suggestion just also need French translation

**Comment by StacyBleeks** [comment]
@thekodester the original PR is fine, the text I provided was simply to add some context. 
do you think it adds value to the PR or maybe not necessary?

**Comment by thekodester** [comment]
Could be added to documentation or as footnote to explain why.
